exec/utils.go

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2024-Present Defense Unicorns
+
+package exec
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"slices"
+	"strings"
+	"sync"
+
+	"github.com/defenseunicorns/pkg/helpers"
+)
+
+var supportedCmdMutations = []string{"zarf", "uds", "kubectl"}
+var registeredCmdMutations = sync.Map{}
+
+// GetFinalExecutablePath returns the absolute path to the current executable, following any symlinks along the way.
+func GetFinalExecutablePath() (string, error) {
+	binaryPath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	// In case the binary is symlinked somewhere else, get the final destination
+	return filepath.EvalSymlinks(binaryPath)
+}
+
+// RegisterCmdMutation registers local ./ commands that should change to the specified cmdLocation
+func RegisterCmdMutation(cmdKey string, cmdLocation string) error {
+	if slices.Contains(supportedCmdMutations, cmdKey) {
+		registeredCmdMutations.Store(cmdKey, cmdLocation)
+		return nil
+	}
+	return fmt.Errorf("%s is not a supported command key", cmdKey)
+}
```

Comment:
unit test this function to ensure that:

- only strings in `supportedCmdMutation` are supported as keys, can be set properly, and return no error
- multiple calls overwrite
- an error is returned if an incorrect `cmdKey` is given

The only current place this function is called does not check this error, nor perform these checks

Reply:
Should this function prevent `""` from being set as a the location?

Reply:
Should this function check if a file exists at the given string and is executable?

Reply:
waiting to unit test until the API settles.